oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -254,15 +261,35 @@ public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bu
                 }
             }
             if (bulkResponse.hasFailures()) { // check if some operations failed to execute
+
+                NodeBuilder status = definitionBuilder.child(IndexDefinition.STATUS_NODE);
+                // Read the current failed paths (if any) on the :status node into failedDocList
+                if (status.getProperty(IndexDefinition.FAILED_DOC_PATHS) != null) {
+                    for (String str : status.getProperty(IndexDefinition.FAILED_DOC_PATHS).getValue(Type.STRINGS)) {
```

Comment:
failedDocPaths isn't a hidden property (which is fine). But because of that, in theory a customer could set it. And he could set it to a single valued property, in which case this would fail (right?). It might be better to protect against such cases... In the past we have seem customers sometimes make mistakes in index definitions. (I'm only asking for protection against _exceptions_ - there is no need to do anything very smart if it's not a multi-valued property).

Reply:
@thomasmueller  - the FAILED_DOC_PATHS property is present on :status node (which is hidden). So I guess that should be ok - right ? Customer would not be able to set the property on the hidden node.